apps/explorer/src/components/owned-objects/views/DynamicFieldsCard.tsx

```diff
@@ -0,0 +1,127 @@
+        observer.observe(element);
+        return () => observer.unobserve(element);
+    }, [fetchNextPage, hasNextPage, handleObserver]);
+
+    if (isInitialLoading) {
+        return (
+            <div className="mt-1 flex w-full justify-center">
+                <LoadingSpinner />
+            </div>
+        );
+    }
+
+    // show the dynamic fields tab if there are pages and the first page has data
+    const hasPages = !!data?.pages?.[0].data.length;
+
+    return hasPages ? (
+        <div className="mt-10">
+            <TabGroup size="lg">
+                <TabList>
+                    <Tab>Dynamic Fields</Tab>
+                </TabList>
+                <TabPanels>
+                    <TabPanel>
+                        <div className="mt-4 flex max-h-[600px] flex-col gap-5 overflow-auto overflow-x-scroll">
+                            {data.pages.map(({ data }) =>
+                                // Show the field name and type is it is not an object
+                                data.map((result) => (
+                                    <DisclosureBox
+                                        title={
+                                            <div className="flex items-center gap-1 truncate break-words text-body font-medium leading-relaxed text-steel-dark">
+                                                {typeof result.name?.value ===
+                                                'object' ? (
+                                                    <div className="block w-8/12 truncate break-words">
+                                                        Struct{' '}
+                                                        {result.objectType}
+                                                    </div>
+                                                ) : (
+                                                    result.name?.value.toString()
+                                                )}
+                                                <ObjectLink
+                                                    objectId={result.objectId}
+                                                />
+                                            </div>
+                                        }
+                                        variant="outline"
+                                        key={result.objectId}
+                                    >
+                                        <div className="flex flex-col divide-y divide-gray-45">
+                                            <UnderlyingObjectCard
+                                                parentId={id}
+                                                name={result.name}
+                                            />
+                                        </div>
+                                    </DisclosureBox>
+                                ))
+                            )}
+
+                            <div ref={observerElem}>
```

Comment:
you can just attach the ref to the loading spinner - i think this is similar enough to what i built for the [coins view](https://github.com/MystenLabs/sui/blob/main/apps/explorer/src/components/OwnedCoins/components/OwnedCoinsPanel.tsx) that you can probably follow the same pattern